lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc

```diff
@@ -382,87 +382,132 @@ function paraneue_dosomething_preprocess_node_campaign(&$vars) {
 
 }
 
+
 /**
- * Adds $gallery_items array of themed Reportback Files for given node preprocess $vars.
+ * Get total number of Reportbacks by status (promoted, approved, etc).
+ *
+ * @param $nid  Campaign node id.
+ * @param $status Reportback status (promoted, approved, etc).
+ * @return int
  */
-function paraneue_dosomething_preprocess_reportback_files(&$vars) {
-  $reportbacks_gallery = &$vars['reportbacks_gallery'];
+function paraneue_dosomething_preprocess_reportbacks_total_by_status($nid, $status) {
+  $params = array();
+  $params['nid'] = $nid;
+  $params['status'] = $status;
 
-  if (!isset($reportbacks_gallery['initial_count'])) {
-    $reportbacks_gallery['initial_count'] = 6;
-  }
-  $initial_count = $reportbacks_gallery['initial_count'];
+  return dosomething_reportback_get_reportback_files_query_count($params);
+}
 
-  $reportbacks_gallery['items'] = array();
 
+/**
+ * Get a specified number of themed Promoted Reportbacks.
+ *
+ * @param $nid  Campaign node id.
+ * @param $count Number of Promoted Reportbacks to retrieve.
+ * @return array
+ */
+function paraneue_dosomething_preprocess_themed_promoted_reportbacks($nid, $count, $random = FALSE) {
+  $promoted_items = array();
 
-  // Promoted Reportbacks:
-  // ========================================================================
-  // Set params to retrieve random Promoted Reportbacks for this nid.
   $params = array();
-  $params['nid'] = $vars['nid'];
+  $params['nid'] = $nid;
   $params['status'] = 'promoted';
-  $params['random'] = TRUE;
-  $promoted_results = dosomething_reportback_get_reportback_files_query_result($params, $initial_count);
+  $params['random'] = $random;
+
+  $promoted_results = dosomething_reportback_get_reportback_files_query_result($params, $count);
 
   foreach ($promoted_results as $rb_file) {
     $rb_file->image = dosomething_image_get_themed_image_url_by_fid($rb_file->fid, '400x400');
-    $reportbacks_gallery['items'][] = paraneue_dosomething_get_gallery_item((array) $rb_file, 'photo');
+    $promoted_items[] = paraneue_dosomething_get_gallery_item((array) $rb_file, 'photo');
   }
 
-  // Total number of Promoted Reportbacks obtained.
-  $reportbacks_gallery['total_promoted'] = count($reportbacks_gallery['items']);
+  return $promoted_items;
+}
 
 
-  // Approved Reportbacks:
-  // ========================================================================
-  // Set params to get the total number of Approved Reportbacks for this nid.
-  $params['status'] = 'approved';
-  if (isset($params['random'])) {
-    unset($params['random']);
-  }
-  $reportbacks_gallery['total_approved'] = dosomething_reportback_get_reportback_files_query_count($params);
-
+/**
+ * Build Reportback collection based on Promoted & Approved Reportback counts.
+ *
+ * @param $reportbacks
+ * @param $desired_count
+ * @param bool $use_placeholders
+ * @return bool
+ */
+function paraneue_dosomething_preprocess_reportback_assimilation ($reportbacks, $use_placeholders = FALSE) {
```

Comment:
LOL. I was being creative with the function name :) Will tone down my creativity, haha.